app/graphql/types/query_type.rb

```diff
@@ -19,14 +19,37 @@ def user(id:)
 
     # Dog Queries
     field :dogs, [Types::DogType], null: false,
-      description: "Return all dogs"
+      description: 'Return all dogs, or all dogs filtered by attribute' do
+        argument :activity_level_range, [Int], required: false
+        argument :age_range, [Float], required: false
+        argument :breed, [String], required: false
+        argument :weight_range, [Int], required: false
+      end
 
-    def dogs
-      Dog.all
+    def dogs(**filters)
+      if filters[:age_range]
+        raise 'Please provide an array with two integers or floating point numbers.' unless filters[:age_range].count == 2
+        filters[:birthdate] = (Time.zone.now - (filters[:age_range].max * 1.year.seconds) - 1.years.seconds)..(Time.zone.now - (filters[:age_range].min * 1.year.seconds))
+        filters.except!(:age_range)
```

Comment:
```suggestion
        filters.delete(:age_range)
```
I think this should do the same thing. If so, let's change them all.

Reply:
Resolved all on branch